### ISSUE TITLE
Discussion: Current State of Release tags

### ISSUE THREAD

**Issue by pancoaster** [projectM-visualizer/projectm#722]
Greetings, 

Dropping in to check on status of the project.  Is the lib ready for re-implementation into UWP frontend?
I do NOT want to implement, then require rework due to lib changes.

Thanks,
coast

**Comment by kblaschke** [comment]
Yes, definitely! We will from now on ensure the API is forward-compatible for all 4.x releases to come, so we'll just add new functions, but won't change or remove existing ones. That said, if you use the 4.0 API, no changes will be required unless you want to migrate to 5.0 at some later point (probably talking about a couple of years here, and even after that we may backport things to 4.x). We've got plenty of features to implement in the current major release, many of them will be available to apps without calling new API functions (yet there will be a few features added requiring new API functions like rendering to a specific FBO, but these are simply improvements which can be adopted at any time).

This was actually one of the two main goals of the 4.0 release:
- Make the build system CMake-based and easy to use
- Replace the C++ "API" of the previous versions with a C API to enable easier use, dynamic linking and disconnecting the actual API from the library implementation details.

Bottom line is, you can start immediately. The 4.1 release will, for example, only contain (huge) internal changes (all-new rendering, parser and expression evaluation code), but keep the API completely untouched, making it a non-effort upgrade. If we introduce new API calls later, you can decide yourself whether you want to use them or not, which solely depends on your requirements.

**Comment by kblaschke** [comment]
Closing the issue, as there's obviously no further need for discussion.

If there is, I'd also suggest posting in the [Discussions](https://github.com/projectM-visualizer/projectm/discussions) section instead, as it's better suited for general questions.
